src/features/drop-manager/routes/nft/[id].tsx

```diff
@@ -1,57 +1,91 @@
-    slug: string;
-    hasClaimed: boolean;
-  }>;
-}
+import { useAuthWalletContext } from '@/contexts/AuthWalletContext';
+import { get } from '@/utils/localStorage';
+import { MASTER_KEY, PAGE_SIZE_LIMIT } from '@/constants/common';
+import { usePagination } from '@/hooks/usePagination';
+import { type DataItem } from '@/components/Table/types';
 
-const tableColumns: ColumnItem[] = [
-  { title: 'Link', selector: (row) => row.link, loadingElement: <Skeleton height="30px" /> },
-  {
-    title: 'Claim Status',
-    selector: (row) => row.hasClaimed,
-    loadingElement: <Skeleton height="30px" />,
-  },
-  {
-    title: '',
-    selector: (row) => row.action,
-    tdProps: {
-      display: 'flex',
-      justifyContent: 'right',
-      verticalAlign: 'middle',
-    },
-    loadingElement: <Skeleton height="30px" />,
-  },
-];
-
-const data: NFTDropResponse = {
-  name: 'Star Invader 3',
-  links: [
-    { id: 1, slug: '#2138h823h', hasClaimed: true },
-    { id: 2, slug: '#2138h823h', hasClaimed: false },
-    { id: 3, slug: '#c34fd2n32', hasClaimed: false },
-    { id: 4, slug: '#rf5hhfaxm', hasClaimed: true },
-  ],
-};
+import { tableColumns } from '../../components/TableColumn';
+import { INITIAL_SAMPLE_DATA } from '../../constants/common';
 
 export default function NFTDropManagerPage() {
+  const { id: dropId } = useParams();
   const [loading, setLoading] = useState(true);
 
-  // TODO: Remove this after backend is ready
+  const [name, setName] = useState('Drop');
+  const [dataSize, setDataSize] = useState<number>(0);
+  const [data, setData] = useState<DataItem[]>([INITIAL_SAMPLE_DATA[0]]);
+
+  const { accountId } = useAuthWalletContext();
+
+  const {
+    hasPagination,
+    pagination,
+    firstPage,
+    lastPage,
+    loading: paginationLoading,
+    handleNextPage,
+    handlePrevPage,
+  } = usePagination({
+    dataSize,
+    handlePrevApiCall: async () => {
+      await handleGetDrops({
+        pageIndex: pagination.pageIndex - 1,
+        pageSize: pagination.pageSize,
+      });
+    },
+    handleNextApiCall: async () => {
+      await handleGetDrops({
+        pageIndex: pagination.pageIndex + 1,
+        pageSize: pagination.pageSize,
+      });
+    },
+  });
+
+  const handleGetDrops = async ({ pageIndex = 0, pageSize = PAGE_SIZE_LIMIT }) => {
+    if (!accountId) return;
+
+    const drop = await getDropInformation({
+      dropId,
+    });
+
+    setDataSize(drop.next_key_id);
+
+    setName(JSON.parse(drop.metadata as string).dropName);
+
+    const { publicKeys, secretKeys } = await generateKeys({
+      numKeys: Math.min(drop.next_key_id, pageSize),
+      rootEntropy: `${get(MASTER_KEY) as string}-${dropId}`,
+      autoMetaNonceStart: pageIndex * pageSize,
+    });
+
+    const keyInfo = await getKeyInformationBatch({
+      publicKeys,
+      secretKeys,
+    });
+
+    setData(
+      secretKeys.map((key, i) => ({
+        id: i,
+        link: 'https://keypom.xyz/claim/' + key.replace('ed25519:', ''),
```

Comment:
@BenKurrek @mattlockyer should we add contract id to this link?
AFAIK, it should like this right? 
'https://keypom.xyz/claim/' + contractId + '/' + key.replace('ed25519:', '')